azuredevops/internal/service/taskagent/resource_variable_group.go

```diff
@@ -338,24 +361,95 @@ func expandVariableGroupParameters(d *schema.ResourceData) (*taskagent.VariableG
 
 	// Note: this will be of length 1 based on the schema definition above.
 	if len(keyVault) == 1 {
-		keyVaultValues := keyVault[0].(map[string]interface{})
+		kvConfigures := keyVault[0].(map[string]interface{})
+		kvName := kvConfigures[vgName].(string)
+		serviceEndpointID := kvConfigures[vgServiceEndpointID].(string)
 
-		serviceEndpointID, err := uuid.Parse(keyVaultValues[vgServiceEndpointID].(string))
+		serviceEndpointUUID, err := uuid.Parse(serviceEndpointID)
 		if err != nil {
 			return nil, nil, err
 		}
 
 		variableGroup.ProviderData = taskagent.AzureKeyVaultVariableGroupProviderData{
-			ServiceEndpointId: &serviceEndpointID,
-			Vault:             converter.String(keyVaultValues[vgName].(string)),
+			ServiceEndpointId: &serviceEndpointUUID,
+			Vault:             &kvName,
 		}
 
 		variableGroup.Type = converter.String(azureKeyVaultType)
-	}
+		if azureKVSecrets, err := azureKVSecrets(clients, *projectID, kvName, serviceEndpointID); err != nil {
+			return nil, nil, err
+		} else {
```

Comment:
This else is redundant due to the return statement in the if. You can just remove the else nesting.